pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go

```diff
@@ -0,0 +1,650 @@
+		return fmt.Errorf("missing TLSAuth secret information")
+	}
+	if v.secrets.DiffieHellmanKey.Name == "" || v.secrets.DiffieHellmanKey.Checksum == "" {
+		return fmt.Errorf("missing DH secret information")
+	}
+	if v.secrets.Server.Name == "" || v.secrets.Server.Checksum == "" {
+		return fmt.Errorf("missing server secret information")
+	}
+
+	var (
+		configMap       = v.emptyConfigMap()
+		serverSecret    = v.emptySecret(DeploymentName)
+		tlsAuthSecret   = v.emptySecret(VpnSeedServerTLSAuth)
+		dhSecret        = v.emptySecret(vpnSeedServerDH)
+		service         = v.emptyService()
+		deployment      = v.emptyDeployment()
+		networkPolicy   = v.emptyNetworkPolicy()
+		gateway         = v.emptyGateway()
+		virtualService  = v.emptyVirtualService()
+		destinationRule = v.emptyDestinationRule()
+	)
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, v.client, serverSecret, func() error {
+		serverSecret.Type = corev1.SecretTypeTLS
+		serverSecret.Data = v.secrets.Server.Data
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, v.client, tlsAuthSecret, func() error {
+		tlsAuthSecret.Type = corev1.SecretTypeOpaque
+		tlsAuthSecret.Data = v.secrets.TLSAuth.Data
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, v.client, dhSecret, func() error {
+		dhSecret.Type = corev1.SecretTypeOpaque
+		dhSecret.Data = v.secrets.DiffieHellmanKey.Data
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, v.client, networkPolicy, func() error {
+		networkPolicy.ObjectMeta.Annotations = map[string]string{
+			v1beta1constants.GardenerDescription: "Allows only Ingress/Egress between the kube-apiserver of the same control plane and the corresponding vpn-seed-server and Ingress from the istio ingress gateway to the vpn-seed-server.",
+		}
+		networkPolicy.Spec = networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					v1beta1constants.GardenRole: v1beta1constants.GardenRoleControlPlane,
+					v1beta1constants.LabelApp:   DeploymentName,
+				},
+			},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{
+				{
+					From: []networkingv1.NetworkPolicyPeer{
+						{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									v1beta1constants.GardenRole: v1beta1constants.GardenRoleControlPlane,
+									v1beta1constants.LabelApp:   v1beta1constants.LabelKubernetes,
+									v1beta1constants.LabelRole:  v1beta1constants.LabelAPIServer,
+								},
+							},
+						},
+					},
+				},
+				{
+					From: []networkingv1.NetworkPolicyPeer{
+						{
+							// we don't want to modify existing labels on the istio namespace
+							NamespaceSelector: &metav1.LabelSelector{},
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									v1beta1constants.LabelApp: "istio-ingressgateway",
```

Comment:
Can you add `TODO (mvladev): use configurable labels`

Reply:
Do you mean as a simple comment or in some special syntax?

Reply:
@ScheererJ Simple comment, ie.., `// TODO(mvladev): ...`

Reply:
Done with latest commit.